### DISCUSSION TITLE
Reality Data Sample Data

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#513] [Q&A]
in Reality Data 、Swiping Comparison sample
the how i can publish the osgb file to the reality data by the ”iTwin Snapshot“ ? and how can the reality data link the bim model ? 
thank you!

**Comment by calebmshafer** [comment]
@Ellord207 @roopksaini  can either of you assist with this how the reality data was published for the Sample?

**Comment by williamkbentley** [comment]
Reality data models are not typically stored in the snapshot iModel due to their large size. They should be stored and served from a different service.

You can attach a reality data model to a snapshot iModel the same way the reality data sample does using [`attachRealityModel`](https://www.itwinjs.org/reference/imodeljs-frontend/views/displaystylestate/attachrealitymodel/):
```
	const props = { tilesetUrl:  any3dTilesetUrl ) };
	vp.attachRealityModel(props);
````

The osgb file will need to be converted to 3D tile format and served by any3dTilesetUrl.

Hope that helps

**Reply by pmconne** [reply]
@williamkbentley [Elsewhere](https://github.com/imodeljs/imodeljs/discussions/643) @paprophet is asking for information on how to do what  you suggest (convert osgb to 3D tiles). Do you have suggestions?
